packages/react-client/src/ReactFlightClient.js

```diff
@@ -197,6 +223,55 @@ function createElement(type, key, props): React$Element<any> {
   return element;
 }
 
+type UninitializedBlockPayload<Data> = [
+  mixed,
+  ModuleMetaData | Chunk<ModuleMetaData>,
+  Data | Chunk<Data>,
+];
+
+type Thenable<T> = {
+  then(resolve: (T) => mixed, reject?: (mixed) => mixed): Thenable<any>,
+};
+
+function initializeBlock<Props, Data>(
+  tuple: UninitializedBlockPayload<Data>,
+): BlockComponent<Props, Data> {
+  // Require module first and then data. The ordering matters.
+  let moduleMetaData: ModuleMetaData = readMaybeChunk(tuple[1]);
+  let moduleReference: ModuleReference<
+    BlockRenderFunction<Props, Data>,
+  > = resolveModuleReference(moduleMetaData);
+  // TODO: Do this earlier, as the chunk is resolved.
+  preloadModule(moduleReference);
```

Comment:
The comment before its implementation says

```
 Returning null means [...]
```

but retvalue is never used. Is it outdated?

Also, is it completely unobservable? I deleted this and nothing failed.

Reply:
The comment is outdated.

It's not unobservable because of this: https://github.com/facebook/react/blob/master/packages/react-flight-dom-webpack/src/ReactFlightClientWebpackBundlerConfig.js#L60-L61

Lots more tests needs to be written at some point.